cmd/antrea-controller/controller.go

```diff
@@ -96,23 +101,64 @@ func run(o *Options) error {
 
 	go apiServer.GenericAPIServer.PrepareRun().Run(stopCh)
 
+	if o.config.EnablePrometheusMetrics {
+		go initializePrometheusMetrics(
+			o.config.EnablePrometheusGoMetrics,
+			o.config.EnablePrometheusProcessMetrics)
+	}
+
 	<-stopCh
 	klog.Info("Stopping Antrea controller")
 	return nil
 }
 
+// Initialize Prometheus metrics collection.
+func initializePrometheusMetrics(
+	enablePrometheusGoMetrics bool,
+	enablePrometheusProcessMetrics bool) {
+	hostname, err := os.Hostname()
+	if err != nil {
+		klog.Errorf("Failed to retrieve agent node name, %v", err)
+	}
+
+	klog.Info("Initializing prometheus")
+	gaugeHost := prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "antrea_controller_host",
+		Help: "Antrea controller hostname (as a label), typically used in grouping/aggregating stats; " +
+			"the label defaults to the hostname of the host but can be overridden by configuration. " +
```

Comment:
I did not see it can be overridden by configuration?

Reply:
initializePrometheusMetrics() is only called when Prometheus metrics are on. Or are you suggesting to override the hostname with a configuration parameter?

Reply:
I am asking your sentence: "the label defaults to the hostname of the host but can be overridden by configuration." - I did not see how we override the label as this sentence states.

Reply:
Ah now I understand - the comment is stale, I'll edit it